src/runtime/markdown-parser/handler/link.ts

```diff
@@ -25,3 +26,15 @@ export default function link (h: H, node: Node) {
 
   return h(node, 'a', props, all(h, node))
 }
+
+function normalizeLink (link: string) {
+  if (isRelative(link) || (!/^https?/.test(link) && !link.startsWith('/'))) {
+    return link.split('/')
+      .map(x => x
+        .replace(/^[0-9]*\./g, '')
+        .replace(/\.md$/g, ''))
+      .join('/')
```

Comment:
```suggestion
      .map(x => x.replace(/^[0-9]*\./g, ''))
      .join('/')
      .replace(/\.md$/g, '')
```

We only need to remove `.md` from the file name